docs/manual/returns-receiver-checker.tex

```diff
@@ -0,0 +1,118 @@
+\htmlhr
+\chapterAndLabel{Returns Receiver Checker}{returns-receiver-checker}
+
+The Returns Receiver Checker enables documenting and checking that a method
+returns its receiver (i.e., the \<this> parameter).
+
+There are two ways to run the Returns Receiver Checker.
+\begin{itemize}
+\item
+Typically, it is automatically run by another checker.
+\item
+Alternately, you can run just the Returns Receiver Checker, by
+supplying the following command-line options to javac:
+\code{-processor org.checkerframework.common.returnsreceiver.ReturnsReceiverChecker}
+\end{itemize}
+
+
+\sectionAndLabel{Annotations}{returns-receiver-checker-annotations}
+
+The qualifier \refqualclass{common/returnsreceiver/qual}{This} on the return
+type of a method indicates that the method returns its receiver.  Methods
+that return their receiver are common in so-called ``fluent'' APIs.  Here
+is an example:
+
+\begin{Verbatim}
+class MyBuilder {
+  @This MyBuilder setName(String name) {
+    this.name = name;
+    return this;
+  }
+}
+\end{Verbatim}
+
+An \refqualclass{common/returnsreceiver/qual}{This} annotation can only be
+written on the return type of a method or in a downcast.
+
+As is standard, the Returns Receiver Checker has a top qualifier,
+\refqualclass{common/returnsreceiver/qual}{UnknownThis}, and a bottom qualifier,
+\refqualclass{common/returnsreceiver/qual}{BottomThis}.
+Pragrammers rarely need to write these annotations.
+
+Here are some implementation details that most users will not care about.
+\refqualclass{common/returnsreceiver/qual}{This}
+is \emph{not} a subtype of \refqualclass{common/returnsreceiver/qual}{UnknownThis},
+but instead is a polymorphic qualifier.  When the Returns Receiver Checker observes an
+\refqualclass{common/returnsreceiver/qual}{This} annotation on a method return type,
+it adds \refqualclass{common/returnsreceiver/qual}{This} to
+the method's receiver type. The presence of the polymorphic annoattion
```

Comment:
This talks about a default that you implemented, it doesn't explain the type hierarchy.
A polymorphic annotation gets resolved to one of the other qualifiers in the hierarchy. The only other qualifiers are `UnknownThis` and `BottomThis`... which one is the default otherwise? You simply reply on the fact that within a method you don't know what a polymorphic qualifier is resolved to and is the only thing that is assignable to a poly return. You don't care at all about polymorphism at the invocation site.
So poly resolution and the hierarchy could be described a bit nicer here...